manifest.go

```diff
@@ -567,6 +570,97 @@ func ReadManifestList(in io.Reader) ([]ManifestFile, error) {
 	return out, dec.Error()
 }
 
+// WriteManifestListV2 writes a list of v2 manifest files to an avro file.
+func WriteManifestListV2(out io.Writer, files []ManifestFile) error {
+	return writeManifestList(out, files, 2)
+}
+
+// WriteManifestListV1 writes a list of v1 manifest files to an avro file.
+func WriteManifestListV1(out io.Writer, files []ManifestFile) error {
+	return writeManifestList(out, files, 1)
+}
```

Comment:
Since we enforce that the version has to match for all of the manifest files, we can avoid needing two separate functions and simply determine our version from the version of the first file in the slice, erroring on an empty slice. 

I think that would be a better API than having separate `WriteManifestListV1` and `WriteManifestListV2` functions.